### PR TITLE
#512

### LINKED ISSUE #512: Add a way to bind the server
Hello,

as PeerTube is setup behind a reverse proxy following the production documentation, it'd seem useful to be able to make it bind to 127.0.0.1:port instead of 0.0.0.0:port.
After searching I didn't find any way of changing that.

Regards,

### PR DESCRIPTION
**Pull request by daftaupe**
Hello,

this is an attempt to fix #512.

Regards,

### CODE REVIEW

File: config/default.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 listen:
+  hostname: ''
```
Comment (Chocobozzz) [inline config/default.yaml]: By default bind on  `127.0.0.1`

File: config/test-1.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 listen:
+  hostname: ''
```
Comment (Chocobozzz) [inline config/test-1.yaml]: We don't need this parameter in test configurations

File: server.ts [hunk 3]
```diff
@@ -214,7 +215,7 @@ async function startApplication () {
   Redis.Instance.init()
 
   // Make server listening
-  server.listen(port)
-  logger.info('Server listening on port %d', port)
+  server.listen(port, hostname)
+  logger.info('Server listening on port %s:%d', hostname, port)
```
Comment (Chocobozzz) [inline server.ts]: on ~port~ %s:%d

**Comment by Chocobozzz** [comment]
Please update docker config too https://github.com/Chocobozzz/PeerTube/blob/develop/support/docker/production/config/production.yaml#L2

File: config/default.yaml (hunk 1 above)
Comment (daftaupe) [reply config/default.yaml]: Alright, I did it for the production example, but I didn't want to break the default config.

File: server.ts (hunk 3 above)
Comment (daftaupe) [reply server.ts]: woops, typo :)

**Comment by daftaupe** [comment]
I will do the updates later, thanks for the review.

File: config/test-1.yaml (hunk 2 above)
Comment (daftaupe) [reply config/test-1.yaml]: Are you sure @Chocobozzz  ? I think that would fail to start in that case.

Comment (Chocobozzz) [reply config/test-1.yaml]: Test configurations inherit the `default` configuration, so if you do https://github.com/Chocobozzz/PeerTube/pull/514#discussion_r181984904 it should work :)

**Comment by Chocobozzz** [comment]
Thanks @daftaupe

**Pull request merged**